src/manifolds/QuotientManifold.jl

```diff
@@ -0,0 +1,88 @@
+
+"""
+    IsQuotientManifold <: AbstractTrait
+
+Specify that a certain decorated Manifold is a quotient manifold in the sence that it provides
+implicitly (or explicitly through [`QuotientManifold`](@ref)  properties of a quotient manifold.
+
+See [`QuotientManifold`](@ref) for more details.
+"""
+struct IsQuotientManifold <: AbstractTrait end
+
+#
+# (Explicit) Quotient manifolds
+#
+@doc raw"""
+    QuotientManifold{M <: AbstractManifold{𝔽}, N} <: AbstractManifold{𝔽}
+
+Equip a manifold ``\mathcal M`` explicitly with the property of being a quotient manifold.
+
+A manifold ``\mathcal M`` is then a a quotient manifold of another manifold ``\mathcal N``,
+i.e. for an [equivalence relation](https://en.wikipedia.org/wiki/Equivalence_relation) ``∼``
+on ``\mathcal N`` we have
+
+```math
+    \mathcal M = \mathcal N / ∼ = \bigl\{ [p] : p ∈ \mathcal N \bigr\},
+```
+where ``[p] ≔ \{ q ∈ \matcal N : q ∼ p\}`` denotes the equivalence class containing ``p``.
+For more details see Subsection 3.4.1[^AbsilMahonySepulchre2008].
+
+This manifold type models an explicit quotient structure.
+This should be done if either the default implementation of ``\mathcal M``
+uses another representation different from the quotient structure or if
+it provides a (default) quotient structure that is different from the one tintroduced here.
```

Comment:
```suggestion
it provides a (default) quotient structure that is different from the one introduced here.
```

Reply:
By the way, I don't really understand this paragraph right now but it should be more clear later when we have some examples :).

Reply:
It means the same as for the metric or embedded manifold, we again have implicit cases (just the Trait) or explicit ones. I am not 100% sure whether we maybe even can omit the ProjectionPoints this way (since Stiefel is the default for Grassmann, then the QuotientManifold of Stiefel could again be just build on matrices),